src/benches/idbstore.rs

```diff
@@ -0,0 +1,98 @@
+use crate::kv::idbstore::IdbStore;
+use crate::kv::Store;
+use rand::Rng;
+use wasm_bench::*;
+
+fn rand_bytes(len: usize) -> Vec<u8> {
+    (0..len).map(|_| rand::random::<u8>()).collect()
+}
+
+fn rand_string(len: usize) -> String {
+    let mut rng = rand::thread_rng();
+    std::iter::repeat(())
+        .map(|_| rng.sample(rand::distributions::Alphanumeric))
+        .take(len)
+        .collect()
+}
+
+#[wasm_bench]
+async fn read1x256(b: &mut Bench) {
+    read1x(b, 256).await
+}
+
+#[wasm_bench]
+async fn read1x1024(b: &mut Bench) {
+    read1x(b, 1 * 1024).await
+}
+
+#[wasm_bench]
+async fn read1x4096(b: &mut Bench) {
+    read1x(b, 4 * 1024).await
+}
+
+#[wasm_bench]
+async fn read1x16384(b: &mut Bench) {
+    read1x(b, 16 * 1024).await
+}
+
+#[wasm_bench]
+async fn read1x65536(b: &mut Bench) {
+    read1x(b, 64 * 1024).await
+}
+
+async fn read1x(b: &mut Bench, size: u64) {
+    let store = IdbStore::new(&rand_string(12)[..]).await.unwrap().unwrap();
+    let key = rand_string(12);
+    let wt = store.write().await.unwrap();
+    wt.put(&key, &rand_bytes(size as usize)).await.unwrap();
+    wt.commit().await.unwrap();
+
+    let rt = store.read().await.unwrap();
+    b.bytes = size;
+    let n = b.iterations();
+    b.reset_timer();
+    for _ in 0..n {
+        rt.get(&key).await.unwrap();
```

Comment:
So in this loop we're reading the same key over and over again right? I wonder if that is telling us what we want to know, since presumably idb is doing some caching internally?

Reply:
Added code to generate a different key for each iteration, which I agree is more realistic: https://github.com/rocicorp/repc/pull/56/commits/87527106c7a7f170cf09ad49691bcb2a563575e9#diff-b55dc3130fafa7cc3087076de344c428R48-R51.